lib/tasks/tiny_mce_asset.rake

```diff
@@ -6,4 +6,75 @@ namespace :tiny_mce_asset do
                        'result_text_id IS ? AND created_at < ?',
                        nil, nil, 7.days.ago)
   end
+
+  desc 'Generate new tiny_mce_assets and replace old assets in RTE'
+  task regenerage_images: :environment do
+    regex = /\[~tiny_mce_id:([0-9a-zA-Z]+)\]/
+    replaced_images = 0
+    failed_attempts = 0
+    all_images = TinyMceAsset.count
+    failed_attempts_ids = []
+    puts 'Start processing steps...'
+    Step.find_each do |step|
+      next unless step.description
+      images = step.description.match(regex)
+      next unless images
+      team = step.protocol.team
+      step.description.gsub!(regex) do |el|
+        match = el.match(regex)
+        old_img = TinyMceAsset.find_by_id(match[1])
+        new_img = TinyMceAsset.create(image: old_img.image,
+                                      team: team,
+                                      reference: step)
+        if new_img
+          # This image will be removed by `remove_obsolete_images` rake task
+          # until all the steps are not updated we still need this image
+          # in case it appears on some other step
+          old_img.update_attributes(team_id: nil, step_id: nil)
+          replaced_images += 1
+          "[~tiny_mce_id:#{new_img.id}]"
+        else
+          failed_attempts += 1
+          failed_attempts_ids << old_img.id
+          "[~tiny_mce_id:#{old_img.id}]" # return the old img
+        end
+      end
+      step.save
+    end
+    puts 'Completed processing steps...'
+    puts 'Start processing result_texts...'
+    ResultText.find_each do |result_text|
+      next unless result_text.text
+      images = result_text.text.match(regex)
+      next unless images
+      team = result_text.result.my_module.protocol.team
+      result_text.text.gsub!(regex) do |el|
+        match = el.match(regex)
+        old_img = TinyMceAsset.find_by_id(match[1])
+        new_img = TinyMceAsset.create(image: old_img.image,
+                                      team: team,
+                                      reference: result_text)
+        if new_img
+          # This image will be removed by `remove_obsolete_images` rake task
+          # until all the steps are not updated we still need this image
+          # in case it appears on some other step
+          old_img.update_attributes(team_id: nil, step_id: nil)
```

Comment:
The team reference should be kept, otherwise the destruction of old images doesn't work (due to team space calculations).

Also, you probably want to set the `result_text_id` to `nil`, not `step_id`...